GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace GenericsHomework
+{
+    public class Node<T> where T : notnull
+    {
+        public Node(T value)
+        {
+            Value = value;
+            Next = this;
+        }
+
+        public T Value { get; private set; }
+        public Node<T> Next { get; private set; }
+
+        public void Append(T value)
+        {
+            if (Exists(value))
+            {
+                throw new ArgumentException("This value already exists");
+            }
+
+            Node<T> node = new Node<T>(value);
+            Node<T> tracer = this;
+
+           while(tracer.Next != this) { tracer = tracer.Next; }
+
+            tracer.Next = node;
+            node.Next = this;
+        }
+
+        public void Clear()
+        {
+            Node<T> prev = this, next = this.Next;
+
+            while(next != this)     //loop that makes all nodes reference themselves.
+            {
+                prev.Next = prev;
+                prev = next;
+                next = next.Next;
+            }
+            prev.Next = prev;
+        }//Garbage collection is not an issue because all nodes that aren't referenced to by main, will have no references after leaving the scope of this method.
+
+        public bool Exists(T value)
+        {
+            if (this.Value.Equals(value)) { return true; }
+
+            Node<T> tracer = this;
+
+            for (; tracer.Next != this; tracer = tracer.Next)
+            {
+                if(tracer.Value.Equals(value)) { return true; }
+            }
+
+            if (tracer.Value.Equals(value)) { return true; }
+
+            return false;
+        }
+
+        public override string? ToString()
+        {
+            if(this.Value is null)
+            {
+                return null;
+            }
+
+            return this.Value.ToString();
```

Comment:
```suggestion
return Value.ToString();
```
The above change will do the same thing.